### ISSUE TITLE
delimiters overwrite parts of adjacent characters

### ISSUE THREAD

**Issue by mnewt** [whitequark/rainbowth#8]
Parts of characters are getting overwritten. 

Example code:

``` clojure
(defn function [mom]
  (map 
    #(make %) [mom dad]))
```

With rainbowth:
![screenshot](https://cloud.githubusercontent.com/assets/1477258/9263865/d29f130e-41d5-11e5-9632-234411b11069.png)

Without rainbowth:
![screenshot](https://cloud.githubusercontent.com/assets/1477258/9263983/d2cbb91c-41d6-11e5-9d5e-a75624abde96.png)

Notice that italics make it worse, but it happens with non-italic characters as well.

On ST3 3083, OSX, MacBook Air 13, relevant settings:

``` json
    "color_scheme": "Packages/User/Monokai.tmTheme",
    "font_face": "Monaco",
    "font_size": 12,
```

**Comment by whitequark** [comment]
I'm afraid there is nothing I can do to fix this. It happens on the intersection of Sublime's rendering engine and the metrics of your font, neither of which I can control in any way.

**Comment by mnewt** [comment]
Ah, I thought that might be the case. Thanks for clarifying.

Do you know why it only happens when rainbowth colors the delimiters?

**Comment by whitequark** [comment]
Try turning subpixel antialiasing off (in Sublime or OS X globally)

**Comment by mnewt** [comment]
Still the same problem. Not as severe, but still there.

rainbowth:
![screenshot](https://cloud.githubusercontent.com/assets/1477258/9265077/f6480290-41e2-11e5-99a7-1422f459225d.png)
No rainbowth:
![screenshot](https://cloud.githubusercontent.com/assets/1477258/9265079/fe761bbe-41e2-11e5-8f2e-c1d48d6b15fd.png)

**Comment by whitequark** [comment]
This doesn't happen at all for me on Linux. Guess it's some kind of OS X-specific bug.

**Comment by mnewt** [comment]
Fair enough. When I have some more time, I'll try and figure out why this is happening and reply back. In the meantime, I'll keep using rainbowth, because even with this issue it's still awesome.

**Comment by mnewt** [comment]
Observe the same issue occurring in other operating systems.

Windows 8.1:
![image](https://cloud.githubusercontent.com/assets/1477258/9398141/b492c3c8-4758-11e5-8d66-d3fbe944007c.png)

Linux (Arch GNOME 3):
![screenshot](https://cloud.githubusercontent.com/assets/1477258/9398169/e36661be-4758-11e5-8946-c40feb2523ff.png)

And without rainbowth:
![screenshot](https://cloud.githubusercontent.com/assets/1477258/9398172/ed840e30-4758-11e5-82b6-b6d87698b8eb.png)

Both of these are using the latest respective beta of Sublime Text 3, 3083, default font settings and default Monokai theme.

**Comment by whitequark** [comment]
Ah. Right. I can reproduce this on my Linux install too.

The problem is that if I leave the background transparent, Sublime draws the border around the letter with the foreground color, instead of changing the color of the letter. My fix to that consisted of extracting the default background color (and for highlighted line too), and setting it explicitly. And some letters, notably `%` or italic `m`, are intersecting with the background that it draws. Since my highlighting schemes do not use italic, I did not notice it initially.

I entirely understand how irritating it is. This should be reported and fixed upstream. However, Sublime's channels to report bugs upstream are really terrible and I have never got an useful response before. So I will leave it to someone with more time and motivation on their hands to triage this issue, because right now I have neither, not for this issue.

**Comment by whitequark** [comment]
It might be possible to rollback to an earlier Sublime version meanwhile, like around 3060 or so. Not satisfactory, I know.

**Comment by skimania** [comment]
+1 for this affects me and I want a fix. Just found the package, which I have been desperately wanting, but don't really think I can use it without this fix.

**Comment by whitequark** [comment]
Try complaining to the developer of Sublime Text. There's nothing I can do about it.
